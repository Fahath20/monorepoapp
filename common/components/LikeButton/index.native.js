import React from 'react';
import { TouchableOpacity, Text } from 'react-native';
import { globalStyles } from '../../styles';
TouchableOpacity.defaultProps = { activeOpacity: 0.8 };

//Basic native button
const AppButton = ({ onPress, title }) => (
	<TouchableOpacity onPress={onPress} style={globalStyles.appButtonContainer}>
	  <Text style={globalStyles.appButtonText}>Like</Text>
	</TouchableOpacity>
  );

export default AppButton;
