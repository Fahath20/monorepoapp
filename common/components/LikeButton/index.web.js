import React, { useState } from "react";
import Heart from "react-animated-heart";

//Fully web supported heart animation
export default function LikeButton() {
  const [isClick, setClick] = useState(false);
  return (
    <div className="LikeButton">
      <Heart isClick={isClick} onClick={() => setClick(!isClick)} />
    </div>
  );
}