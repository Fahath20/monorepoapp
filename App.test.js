import React from 'react';
import renderer from 'react-test-renderer';
import LikeButton from './common/components/LikeButton/index.native';

test('renders correctly', () => {
  const tree = renderer.create(<LikeButton />).toJSON();
  expect(tree).toMatchSnapshot();
});