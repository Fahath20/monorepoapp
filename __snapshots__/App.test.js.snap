// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#0066AE",
      "borderRadius": 10,
      "elevation": 8,
      "opacity": 1,
      "paddingHorizontal": 12,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#fff",
        "fontSize": 18,
        "fontWeight": "bold",
        "textTransform": "uppercase",
      }
    }
  >
    Like
  </Text>
</View>
`;
